src/components/SampleTable.tsx

```diff
@@ -58,17 +61,14 @@ interface Props {
 
 // SampleTable shows detailed information about individual samples from GISAID
 export const SampleTable = ({ matchPercentage, variant, country, samplingStrategy }: Props) => {
-  const [samples, setSamples] = useState<Sample[] | undefined>(undefined);
-  const [totalNumber, setTotalNumber] = useState<number | undefined>(undefined);
-
-  useEffect(() => {
-    let isSubscribed = true;
+  const mutationsString = variant.mutations.join(',');
+  const { isLoading, isSuccess, error, isError, data: samples, refetch, isFetching } = useQuery<
+    SampleResultList,
+    Error
+  >('samples', () => {
```

Comment:
Hi @ningxie1991. I just started to read about React query. Am I understanding it correctly that the query key should always contain all variables, i.e., in this case the pangolin lineage, the mutations string, the match percentage, the country, and the sampling strategy?

https://react-query.tanstack.com/guides/query-keys

Reply:
The query key is an identifier for the particular api request. The parameters for the api request are given in the query function right below:

https://github.com/cevo-public/cov-spectrum-website/blob/f82638d8c3ee4093898dac5bbfd57fd8535ee1ee/src/components/SampleTable.tsx#L72-L78

Reply:
Oh I see what you mean. We should add the dependent variables to the query key as well.